extensions/iceberg/src/main/java/io/deephaven/iceberg/layout/IcebergBaseLayout.java

```diff
@@ -140,6 +141,9 @@ public synchronized void findKeys(@NotNull final Consumer<IcebergTableLocationKe
         try {
             // Retrieve the manifest files from the snapshot
             final List<ManifestFile> manifestFiles = snapshot.allManifests(table.io());
+            // Sort manifest files by sequence number to read data files in the correct commit order
+            manifestFiles.sort(Comparator.comparingLong(ManifestFile::sequenceNumber));
+            // TODO(deephaven-core#5989: Add unit tests for the ordering of manifest files
```

Comment:
I have a lot of unit tests in #5989, so not adding here since its harder to unit test without the ability to append multiple data files in a single snapshot.